scripts/docker/build_docker.py

```diff
@@ -276,28 +308,29 @@ def get_build_priority(self, target_name: str) -> (int, str):
                 else 1 + max((self.get_build_priority(build_dep)[0] for build_dep in build_deps.keys()), default=0)
         return self.build_priority[target_name], target_name
 
-    def _add_image_prereqs(self, build_targets: Set[str]) -> Set[str]:
+    def _add_image_prereqs(self, registry: ContainerRegistry, build_targets: Set[str]) -> Set[str]:
         # Ensure that image prerequisite that a target requires exists. If not, add them to targets to build.
         # (This should almost never happen unless someone has messed with dockers.json or added a brand-new image.)
         prereqs = {
             prereq
             for target in build_targets
             for prereq in self.dependencies[target].docker_dependencies.keys()
-            if self.get_current_image(prereq, throw_error_on_no_image=False) is None
+            if self.get_current_image(registry, prereq, throw_error_on_no_image=False) is None
```

Comment:
I think this can be made less brittle by passing all the registries (rather than the first one in iteration order) and doing this
```suggestion
            if any(self.get_current_image(registry, prereq, throw_error_on_no_image=False) is None for registry in registries)
```
The logic being that if any registry is missing this image, it needs to be pushed. But if there's a local version of the desired image already built, it won't be passed in (it will me missing from `build_targets`) so it won't be needlessly built. So this allows people to do some naughty things and accidentally get the registries out of sync, then fix them, rather than having someone have to come in and do it manually. The cleverest thing would be to have the code notice that one registry had the built image and the other didn't, and then pull the image to make it local. But that is more complexity.

Reply:
That is interesting. It requires additional refactoring to the `registries` in that method; hence, I prefer to exclude it from this PR as it does it breaks basic functionality as-is.

Reply:
Not sure I follow. Are you saying the code suggestion I make breaks functionality? I don't see how: all it would do is potentially add additional images to the list of build targets (and again, only if they're not already built locally).

Reply:
I apologize for the confusion. To clarify, I meant that the current implementation seems to maintain the basic existing functionality and not causing any new bugs. While implementing your suggestion could improve the code, it would require additional refactoring. Given that the current implementation is not introducing any new bugs, I think it would be better to keep it as is for now.